File: nj2k/testData/newJ2k/inheritance/openModifier.java
import java.lang.Override;
import java.lang.Void;

class A {
    public void f1(){}
    public void f2(){}
    private void f3(){}
}

class B extends A {
    @Override
    public void f1() {
        super.f1();
    }
}

class C extends B {
    @Override
    public void f1() {
        super.f1();
    }
}

interface I {
    void f();
}

class D implements I {
    @Override
    public void f() { }
}

abstract class E {
    abstract void f1();
    void f2(){}
    void f3(){}
}

class F extends E {
    @Override
    void f1() {
    }

    @Override
    void f2() {
        super.f2();
    }
}

File: nj2k/testData/newJ2k/inheritance/openModifier.kt
internal open class A {
    open fun f1() {}
    fun f2() {}
    private fun f3() {}
}

internal open class B : A() {
    override fun f1() {
        super.f1()
    }
}

internal class C : B() {
    override fun f1() {
        super.f1()
    }
}

internal interface I {
    fun f()
}

internal class D : I {
    override fun f() {}
}

internal abstract class E {
    internal abstract fun f1()
    internal open fun f2() {}
    fun f3() {}
}

internal class F : E() {
    public override fun f1() {}
    public override fun f2() {
        super.f2()
    }
}
